src/lib/price.ts

```diff
@@ -4,10 +4,17 @@ import { type Chain, toDefiLlamaChain } from '@lib/chains'
 import { mulPrice, sum } from '@lib/math'
 import type { Token } from '@lib/token'
 import { isNotNullish } from '@lib/type'
+import { formatUnits } from 'viem'
+
+// Defillama prices API requires a prefix to know where the token comes from.
+// It can be a chain or a market provider like coingecko
+export function getTokenKey(token: { chain: Chain; address: string }): string {
+  if (!token.chain || !token.address) {
+    console.error('[getTokenKey]: Invalid token', token)
+    throw new Error('Invalid token')
```

Comment:
please return `null` or `undefined` instead of throwing an error here

Reply:
I throw here because all tokens should have chain and address. If a token doesn't have both props then something is wrong and the process shouldn't continue. Instead it throws and I fix the token. Wdyt?